### PR TITLE
Move model_config to common

### PR DESCRIPTION
**Pull request by kpedro88**
Addresses https://github.com/triton-inference-server/server/issues/2759: many of the functions in `model_config.h` are generically useful even on the client side and only depend on the generated protobuf headers. Therefore, it is useful to have them more broadly accessible via the common repo.

The header and cc files are copied here from the core repo (except for a few functions with non-trivial dependencies that stay in core), with the namespace changed to `triton::common`. A new library `triton-common-model-config` is built. Because many of the functions depend on the generated protobuf headers, this library is only built when the protobuf part of this repo is enabled.

This PR goes with https://github.com/triton-inference-server/core/pull/60.

### CODE REVIEW

**Review by deadeyegoodwin — DISMISSED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -202,6 +202,7 @@ target_include_directories(
 
 target_link_libraries(triton-common-table-printer PRIVATE common-compile-settings)
 
+
```
Comment (jbkyang-nvi) [inline CMakeLists.txt:205]: extra line

**Review by jbkyang-nvi — DISMISSED** [review]

File: include/triton/common/model_config.h [hunk 2]
```diff
@@ -0,0 +1,244 @@
+// Copyright 2018-2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (GuanLuo) [inline include/triton/common/model_config.h:1]: Update copyright here and .cc

**Review by GuanLuo — DISMISSED** [review]
LGTM, just need to update copyright year

**Comment by kpedro88** [comment]
@GuanLuo done

File: src/model_config.cc [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2018-2020, NVIDIA CORPORATION. All rights reserved.
+// Copyright (c) 2018-2022, NVIDIA CORPORATION. All rights reserved.
```
Comment (GuanLuo) [inline src/model_config.cc:1]: Sorry for missing this spot and not being clear, the new format for copyright is the one in model_config.h
```
// Copyright 2018-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

**Review by GuanLuo — APPROVED** [review]

**Comment by kpedro88** [comment]
thanks for handling the last comment @jbkyang-nvi

**Pull request merged**